tests/functional/test_plugins_multi_state_set_unset.py

```diff
@@ -309,5 +326,9 @@ def _test_successful_multi_state_set_unset(
 def _run_successful_task(action: Action, action_owner_name: str):
     """Run action and expect it succeeds"""
     task = action.run()
-    with catch_failed(TaskFailed, f'Action {action.name} should have succeeded when ran on {action_owner_name}'):
-        task.try_wait()
+    try:
+        wait_for_task_and_assert_result(task, status="success")
+    except AssertionError as error:
+        raise AssertionError(
+            f'Action {action.name} should have succeeded when ran on {action_owner_name}:\n' f'{error}'
+        ) from error
```

Comment:
maybe move it to plugin_utils?

Reply:
move it in your pull request, since now it is only needed in this file

Reply:
Sure